### PR TITLE
Fwed

### PR DESCRIPTION
**Pull request by leadheadfred**
looks good ig?

### CODE REVIEW

**Comment by leadheadfred** [comment]
I expect number_of_autos in method signature will mess something up but number of autos important for things like exe, would be fun to make table/graphs of damage eventually but gotta learn more for that

File: damage_calculator.py [hunk 1]
```diff
@@ -335,12 +349,54 @@ def calc_dps(client, god, build, enemy, enemy_build, enemy_level, level=20):
...
+
+        x_axis.append(i)
+        y_axis.append(dmg)
+    plt.plot(x_axis, y_axis)
+    plt.xlabel("Auto Attack Numer")
+    plt.ylabel("Damage Dealt")
+    plt.title("Damage Graph")
+    plt.show()
+
+# store this somewhere, allow user to compare graphs after selecting a different build? just plot 2 sets of numbers
```
Comment (MattK4517) [inline damage_calculator.py]: the plan is to have all the front end data output go to the website, graphing / data output isnt needed in the frontend. All this code should output is json that can be parsed later

File: damage_calculator.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import pymongo
 from main import client
 import analyze as anlz
+import matplotlib as plt
```
Comment (MattK4517) [inline damage_calculator.py]: dont need matplotlib here, data visualization will be done at a later stage

File: damage_calculator.py [hunk 3]
```diff
@@ -180,9 +181,9 @@ def calc_dps_stats(client, god, build, baseAttSpeed):
     for item in build:
         stat = get_special_item(item)
         if "Physical Armor Reduction" in stat.keys() and item == "The Executioner":
-            armor_reduction_per = 7
+            armor_reduction_per = 7 # dont think this is needed as 295 takes exe into account?
         itemcol = itemdb[item]
-        for x in itemcol.find({}, {"ItemDescription": 1}):
+        for x in itemcol.find({}, {"ItemDescription": 1}): #pls update
```
Comment (MattK4517) [inline damage_calculator.py]: what needs to be updated about this?

File: damage_calculator.py [hunk 4]
```diff
@@ -180,9 +181,9 @@ def calc_dps_stats(client, god, build, baseAttSpeed):
     for item in build:
         stat = get_special_item(item)
         if "Physical Armor Reduction" in stat.keys() and item == "The Executioner":
-            armor_reduction_per = 7
+            armor_reduction_per = 7 # dont think this is needed as 295 takes exe into account?
```
Comment (MattK4517) [inline damage_calculator.py]: yea i think you can take and _item == "The Executioner"_ out of the if statement

File: damage_calculator.py [hunk 5]
```diff
@@ -251,11 +252,13 @@ def calc_auto_dmg(god, power):
 
     return attDamage
 
-
-def calc_dps(client, god, build, enemy, enemy_build, enemy_level, level=20):
+x_axis = []
```
Comment (MattK4517) [inline damage_calculator.py]: dont need arrays of datapoints

File: damage_calculator.py [hunk 6]
```diff
@@ -289,20 +292,31 @@ def calc_dps(client, god, build, enemy, enemy_build, enemy_level, level=20):
                  "Ichaival": 0, "Silverbranch Bow": 0, "Total Item Damage": 0}
 
     flag = False
-    for i in range(10):
+    for i in range(number_of_autos):
```
Comment (MattK4517) [inline damage_calculator.py:332]: honestly this is probably better anyways

File: damage_calculator.py [hunk 7]
```diff
@@ -289,20 +292,31 @@ def calc_dps(client, god, build, enemy, enemy_build, enemy_level, level=20):
...
         if "The Executioner" in build:
-            if i <= 4:
+            if i < 5 and i > 0:
```
Comment (MattK4517) [inline damage_calculator.py:334]: test to make sure this functions the same in game, iirc in testing it functions correctly with i <= 4

File: damage_calculator.py [hunk 8]
```diff
@@ -313,14 +327,14 @@ def calc_dps(client, god, build, enemy, enemy_build, enemy_level, level=20):
                 elif "Spectral Armor" in enemy_build and "Deathbringer" in build:
                     dmg += attDamage * (1.3 - .55)
                 else:
-                    dmg += attDamage
+                    dmg = attDamage * 1.75
```
Comment (MattK4517) [inline damage_calculator.py]: dmg is total damage, this will reset it and give incorrect results

File: damage_calculator.py [hunk 9]
```diff
@@ -313,14 +327,14 @@ def calc_dps(client, god, build, enemy, enemy_build, enemy_level, level=20):
...
                 if "Wind Demon" in build and not flag:
                     pen_per += 10
                     flag = True
 
         if (i > 0 and i % 4 == 0) and "Odysseus' Bow" in build:
-            item_dmg += round(15 + (power * .6))
-            item_dmg_out["Odysseus' Bow"] += round(15 + (power * .6))
+            item_dmg += round(15 + (calc_auto_dmg(god, attDamage) * .6))
```
Comment (MattK4517) [inline damage_calculator.py:373]: verify this works correctly in game

File: damage_calculator.py (hunk 3 above)
Comment (leadheadfred) [reply damage_calculator.py]: there is an X what is X >:(

File: damage_calculator.py (hunk 7 above)
Comment (leadheadfred) [reply damage_calculator.py:334]: i think with i<=4 means that the first auto gets exe shred, but the first auto only applies the exe shred, so the second one gets the benefit, so for i>0 and i<5, first pen is applied after first auto, but not sure how to test cause dk how to run it heh

File: damage_calculator.py (hunk 9 above)
Comment (leadheadfred) [reply damage_calculator.py:373]: verified ody bow is based off atk damage not power if thats what you mean

File: damage_calculator.py (hunk 3 above)
Comment (MattK4517) [reply damage_calculator.py]: this is just working with the database the itemcol.find() searches the database and returns an array of data. x in this case would be an array of ItemDescriptions

File: damage_calculator.py (hunk 7 above)
Comment (MattK4517) [reply damage_calculator.py:334]: yea but does the shred get applied before dmg in game, giving the first auto the 7% shred. im fairly sure it does, youd have to check in jungle practice

File: damage_calculator.py (hunk 9 above)
Comment (MattK4517) [reply damage_calculator.py:373]: youd have to check in jungle practice to see which way is correct in terms of damage numbers

File: damage_calculator.py [hunk 10]
```diff
@@ -251,11 +251,11 @@ def calc_auto_dmg(god, power):
 
     return attDamage
 
-
-def calc_dps(client, god, build, enemy, enemy_build, enemy_level, level=20):
+def calc_dps(client, god, build, enemy, enemy_build, enemy_level, number_of_autos, level=20):
```
Comment (MattK4517) [inline damage_calculator.py]: can you set the default number_of_autos to 10

number_of_autos=10 in this line

File: damage_calculator.py (hunk 9 above)
Comment (leadheadfred) [reply damage_calculator.py:373]: item desc says basic attack damage which is good enough imo

File: damage_calculator.py (hunk 7 above)
Comment (leadheadfred) [reply damage_calculator.py:334]: checked in game, shred applied after first auto

File: damage_calculator.py (hunk 9 above)
Comment (leadheadfred) [reply damage_calculator.py:373]: checked in game, applies BA dmg

File: damage_calculator.py (hunk 3 above)
Comment (leadheadfred) [reply damage_calculator.py]: changed x to be item_description

File: damage_calculator.py (hunk 10 above)
Comment (leadheadfred) [reply damage_calculator.py]: donee

**Pull request merged**